deploy/tests/conftest.py

```diff
@@ -83,3 +84,15 @@ def patch(self, task_name: str, wrap_task=False, wrap_config=False) -> firesim.T
             return t
 
     return TaskMocker(mocker)
+
+
+@pytest.fixture
+def mock_paramiko_ssh_and_key():
+    sample_key = os.path.join(os.path.dirname(os.path.abspath(mockssh.__file__)), "sample-user-key")
+
+    with mockssh.Server(users={"sample-user": sample_key}) as server:
+        with server.client(uid="sample-user") as c:
+            sftp = c.open_sftp()
+            sftp.put(os.path.join(os.path.dirname(__file__), "fsspec_test_json.json"), "/tmp/test_file.json", confirm=True)
```

Comment:
It may make sense to eventually refactor this part out into another fixture that builds on top of the base `mock_ssh_server` so that different tests can have different server contents.  Not sure that we need to do that for this initial PR.  Just a thought that I missed before copying the PR into a local branch for testing.

Reply:
Yeah it'd be better to build a fixture that returns a function which actually launches the mock server. The function could take arguments that dictate server contents.